datajunction-clients/python/datajunction/builder.py

```diff
@@ -119,6 +119,7 @@ def create_source(  # pylint: disable=too-many-arguments
         primary_key: Optional[List[str]] = None,
         tags: Optional[List[Tag]] = None,
         mode: Optional[models.NodeMode] = models.NodeMode.PUBLISHED,
+        if_exists: Optional[str] = "update",
```

Comment:
Should this be an enum instead of a str? Then it'll be a bit clearer what the allowed values for this are without going into the code here

Reply:
I would call the param `update_if_exists` and make it a boolean. And also then... I think the name of the function `create_source` could stay as is.